### ISSUE TITLE
.all() not threadsafe since 0.18

### ISSUE THREAD

**Issue by josephlbarnett** [willowtreeapps/assertk#224]
We have tests that, for example, look like:

```
assertThat(string).all {
    contains("thing")
    doesNotContain("thing2")
}
```

and are running them via TestNG and the surefire-maven plugin with parallel=methods, and 10 test threads.
With assertk 0.18, we're seeing the following stack trace when running tests:
```
kotlin.KotlinNullPointerException
	at assertk.FailureContext.pushFailure(failure.kt:16)
	at assertk.Failure$DefaultImpls.pushFailure(failure.kt:57)
	at assertk.SoftFailure.pushFailure(failure.kt:105)
	at assertk.AssertKt.all(assert.kt:339)
	at assertk.AssertKt.all(assert.kt:249)
	at com.trib3.json.ObjectMapperTest.testYearQuarterMap(ObjectMapperTest.kt:81)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:584)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:172)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:804)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:145)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

setting the number of threads down to 1 appears to avoid these failures, but slows down our builds.

**Comment by evant** [comment]
Hm, interesting. It's very likely that soft assertions have never properly been thread-safe. Might be more likely to trigger since there is more inlining in that codepath?

**Comment by evant** [comment]
@josephlbarnett Can you try out 0.19-SNAPSHOT and see if it fixes your issue?

```groovy
repositories {
  maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
}

dependencies {
  testCompile 'com.willowtreeapps.assertk:assertk-jvm:0.19-SNAPSHOT'
}
```

**Comment by josephlbarnett** [comment]
looks good in our initial tests, thanks @evant !

**Comment by evant** [comment]
Released v0.19 with this fix
